src/prefect/tasks.py

```diff
@@ -284,7 +284,7 @@ def with_options(
             ),
             cache_result_in_memory=(
                 cache_result_in_memory
-                if cache_result_in_memory is None
+                if cache_result_in_memory is not None
```

Comment:
I swear I stared at this line for a while in the last PR thinking something was wrong, but convinced myself it was right. Good catch!